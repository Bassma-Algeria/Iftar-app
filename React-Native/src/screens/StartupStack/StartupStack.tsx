import React from 'react';
import {
  createNativeStackNavigator,
  NativeStackNavigationOptions,
} from '@react-navigation/native-stack';

import type {StartupStackStackParamsList} from './StartupStack.types';
import type {MainStackScreenProps} from '../MainStack.types';

import {Welcom} from './screens/Welcom/Welcom';

interface Props extends MainStackScreenProps<'StartupStack'> {}

const Stack = createNativeStackNavigator<StartupStackStackParamsList>();

const StartupStack: React.FC<Props> = () => {
  const screensOptions: NativeStackNavigationOptions = {
    headerShown: false,
  };

  return (
    <Stack.Navigator screenOptions={screensOptions}>
      <Stack.Screen name="Welcom" component={Welcom} />
    </Stack.Navigator>
  );
};

export {StartupStack};
