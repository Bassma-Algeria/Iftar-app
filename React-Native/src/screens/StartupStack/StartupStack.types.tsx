import {NativeStackScreenProps} from '@react-navigation/native-stack';

export type StartupStackStackParamsList = {
  Welcom: undefined;
  ChooseUsageType: undefined;
};

export type StartupStackScreenProps<
  CurrentScreen extends keyof StartupStackStackParamsList,
> = NativeStackScreenProps<StartupStackStackParamsList, CurrentScreen>;
