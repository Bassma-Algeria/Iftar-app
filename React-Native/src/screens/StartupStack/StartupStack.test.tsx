import React from 'react';
import renderer from 'react-test-renderer';

import {StartupStack} from './StartupStack';

describe('<StartupStack />', () => {
  it.skip('should render correctly', () => {
    const defaultProps: any = {};
    const tree = renderer.create(<StartupStack {...defaultProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
