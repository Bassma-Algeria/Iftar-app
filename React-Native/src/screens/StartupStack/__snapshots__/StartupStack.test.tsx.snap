// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartupStack should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Splash"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTScrollView
          style={
            Array [
              Object {
                "backgroundColor": "#F4E6CA",
                "height": "100%",
                "paddingTop": undefined,
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                  "position": "absolute",
                  "top": -30,
                  "width": "100%",
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "maxWidth": 200,
                  }
                }
              />
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "maxWidth": 200,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "marginTop": "60%",
                  "padding": 15,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "paddingTop": 100,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "baseline",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#7C4908",
                            "fontFamily": "Cairo-Bold",
                            "fontSize": 35,
                          },
                          undefined,
                        ]
                      }
                    >
                      إفطار
                    </Text>
                    <Image
                      collapsable={false}
                      nativeID="animatedComponent"
                      resizeMode="contain"
                      source={
                        Object {
                          "process": [Function],
                        }
                      }
                      style={
                        Object {
                          "height": 20,
                          "left": 4,
                          "position": "absolute",
                          "top": -3,
                          "transform": Array [
                            Object {
                              "rotate": "0deg",
                            },
                          ],
                          "width": 20,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginTop": 50,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "backgroundColor": "#CA8B32",
                        "borderRadius": 10,
                        "height": 15,
                        "marginLeft": 15,
                        "marginRight": 15,
                        "transform": Array [
                          Object {
                            "scale": 1,
                          },
                        ],
                        "width": 15,
                      }
                    }
                  />
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "backgroundColor": "#CA8B32",
                        "borderRadius": 10,
                        "height": 15,
                        "marginLeft": 15,
                        "marginRight": 15,
                        "transform": Array [
                          Object {
                            "scale": 1,
                          },
                        ],
                        "width": 15,
                      }
                    }
                  />
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "backgroundColor": "#CA8B32",
                        "borderRadius": 10,
                        "height": 15,
                        "marginLeft": 15,
                        "marginRight": 15,
                        "transform": Array [
                          Object {
                            "scale": 1,
                          },
                        ],
                        "width": 15,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
